noise/README.md

```diff
@@ -217,6 +217,8 @@ When decrypted, the payload contains a serialized [protobuf][protobuf]
 `NoiseHandshakePayload` message with the following schema:
 
 ``` protobuf
+syntax = "proto2";
+
 message NoiseHandshakePayload {
   bytes identity_key = 1;
```

Comment:
```suggestion
  required bytes identity_key = 1;
```

If I am not mistaken `proto2` requires either `required`, `optional` or `repeated`.

https://developers.google.com/protocol-buffers/docs/proto#specifying-rules

Reply:
Any objections against making them all optional? If we ever want to send early data with the first handshake message, we can then use this Protobuf.

Reply:
For the record, a proto3 encoder will simply provide the default value to the user in case any of these fields are `optional` and are not set before encoding with a proto2 encoder.

I am fine with these fields being `optional`. Though keep in mind that that would allow the above incompatibility at the schema level. I.e. a proto2 implementation could send a message to an old proto3 implementation that is incompatible.

How about making them `optional` and  documenting that some implementations (using proto3) expect these fields to be set on the second and third Noise handshake message?

Reply:
> How about making them optional and documenting that some implementations (using proto3) expect these fields to be set on the second and third Noise handshake message?

I think our handshake description specifies exactly that. Not sure if we need to add any text, mind suggesting something if you feel strongly?
Sending this Protobuf in the first handshake is forbidden by this spec anyway (although we don't enforce it), so using this Protobuf for that purpose would require a spec change anyway.

Reply:
> I think our handshake description specifies exactly that.

Yes, you are right. Sorry for that.